crepecake/src/main/java/com/google/cloud/tools/crepecake/cache/Cache.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.crepecake.cache;
+
+import com.google.cloud.tools.crepecake.cache.json.CacheMetadataTemplate;
+import com.google.cloud.tools.crepecake.json.JsonTemplateMapper;
+import com.google.common.annotations.VisibleForTesting;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.NotDirectoryException;
+import java.nio.file.Path;
+
+/** Manages the cache. */
+public class Cache {
+
+  /** The path to the root of the cache. */
+  private final Path cacheDirectory;
+
+  /** The metadata that corresponds to the cache at {@link #cacheDirectory}. */
+  private final CacheMetadata cacheMetadata;
+
+  /**
+   * Initializes a cache with a directory. This also loads the cache metadata if it exists in the
+   * directory.
+   */
+  public static Cache init(File cacheDirectory)
+      throws NotDirectoryException, CacheMetadataCorruptedException {
+    if (!cacheDirectory.isDirectory()) {
+      throw new NotDirectoryException("The cache can only write to a directory");
+    }
+
+    CacheMetadata cacheMetadata = loadCacheMetadata(cacheDirectory.toPath());
```

Comment:
would it make sense to make cacheDirectpry a Path instead of a File?

Reply:
If `cacheDirectory` is a `Path`, then the check for `isDirectory` would need to first convert it to `File`. Is there a standard as to whether `Path` or `File` is preferred for a parameter?

Reply:
No, you can use `Files.isDirectory(path)` instead. 

Generally we prefer Path for new code, but there's a lot of legacy code that will require a File.